### PR TITLE
[Microsoft] support auth-reset command

### PR DESCRIPTION
**Pull request by michal-dagan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
Related PR: [link to the issue](https://github.com/demisto/content/pull/26482)

## Description
Added the `!<integration command prefix>-auth-reset` command to:
- Microsoft Defender for Cloud Event Collector
- Microsoft Defender for Cloud Apps Event Collector
- Microsoft 365 Defender Event Collector
- EWSO365

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftDefenderEventCollector/MicrosoftDefenderEventCollector.py [hunk 1]
```diff
@@ -55,14 +56,15 @@ def load_json(v: Any) -> dict:
...
+    return None
 
 
 class IntegrationHTTPRequest(BaseModel):
     method: Method
     url: AnyUrl
     verify: bool = True
-    headers: dict = dict()  # type: ignore[type-arg]
-    auth: Optional[HTTPBasicAuth]
+    headers: dict = {}  # type: ignore[type-arg]
```
Comment (bziser) [inline Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftDefenderEventCollector/MicrosoftDefenderEventCollector.py:66]: I don't think we need this ignore

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/Microsoft365DefenderEventCollector/Microsoft365DefenderEventCollector.py [hunk 2]
```diff
@@ -55,14 +56,15 @@ def load_json(v: Any) -> dict:
...
     if isinstance(v, dict):
         return v
+    return None
 
 
 class IntegrationHTTPRequest(BaseModel):
     method: Method
     url: AnyUrl
     verify: bool = True
     headers: dict = {}  # type: ignore[type-arg]
```
Comment (bziser) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/Microsoft365DefenderEventCollector/Microsoft365DefenderEventCollector.py:66]: check this ignore too :)

**Review by bziser — COMMENTED** [review]
looks good :)

**Review by bziser — APPROVED** [review]

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftDefenderEventCollector/MicrosoftDefenderEventCollector.py [hunk 3]
```diff
@@ -186,7 +188,6 @@ def get_last_run(events: list) -> dict:
     @abstractmethod
     def _iter_events(self):
         """Create iterators with Yield"""
-        pass
```
Comment (michal-dagan) [inline Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftDefenderEventCollector/MicrosoftDefenderEventCollector.py:189]: raise NotImplementedError

**Pull request merged**